### PR TITLE
Use correct hyperparameters for 4000-label CIFAR-10

### PR DESCRIPTION
**Pull request by tarvaina**
When cleaning the private mean-teacher code for publishing in GitHub, I had accidentally changed the hyperparamters for 4000-label CIFAR-10 in the PyTorch code. 

This PR contains the correct hyperparameters and a couple of small fixes.

### CODE REVIEW

**Comment by tarvaina** [comment]
The incorrect hyperparameters give test error of about 7.3% and the correct ones about 6.3%. Thank you to Thomas Robert for pointing out the discrepancy.
